md-dao/src/main/java/com/wipro/ats/bdre/md/dao/GeneralConfigDAO.java

```diff
@@ -125,12 +126,12 @@ public void delete(GeneralConfigId generalConfigId) {
             session.beginTransaction();
 
             if (required == 2) {
-                Criteria getGeneralConfigCriteria = session.createCriteria(GeneralConfig.class).add(Restrictions.eq("id.configGroup", configGroup)).add(Restrictions.eq("enabled", true));
+                Criteria getGeneralConfigCriteria = session.createCriteria(GeneralConfig.class).add(Restrictions.eq(CONFIG_GROUP, configGroup)).add(Restrictions.eq("enabled", true));
                 List<GeneralConfig> jpaGeneralConfigList = getGeneralConfigCriteria.list();
                 for (GeneralConfig jpaGeneralConfig : jpaGeneralConfigList) {
                     com.wipro.ats.bdre.md.beans.table.GeneralConfig generalConfig = new com.wipro.ats.bdre.md.beans.table.GeneralConfig();
                     generalConfig.setCounter(jpaGeneralConfigList.size());
-                    if (jpaGeneralConfig.getRequired() == true)
+                    if (jpaGeneralConfig.getRequired())
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS134)